### PR TITLE
Isaac plugin vector field vis compatility and benchmarking

### PR DESCRIPTION
**Pull request by FelixTUD**
- Added compatibility to the isaac vector field visualization and usage of vector field sources
- Benchmarking code example, can be enabled in .cfg files with the parameter `--isaac.timingsFilename [PATH/FILENAME.csv]`

Known issues:
- Moving window creates artifacts (for LIC visualization) at front GPUs as the seeding positions for the vector field visualization are stationary

- [x] Requires the picongpu PR https://github.com/ComputationalRadiationPhysics/picongpu/pull/3718 to be merged
- [x] Requires the isaac PR https://github.com/ComputationalRadiationPhysics/isaac/pull/143 to be merged

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
@FelixTUD I added the documentation for the new command-line option and updated the CI to pull the latest ISAAC dev.

@PrometheusPi Could you do a review, at least the commit with the documentation change.

**Review by psychocoderHPC — DISMISSED** [review]

**Comment by PrometheusPi** [comment]
@psychocoderHPC I will do ...

File: docs/TBG_macros.cfg [hunk 1]
```diff
@@ -288,6 +288,8 @@ TBG_restartLoop="--checkpoint.restart.loop 5000"
 #   By default the ISAAC Plugin tries to reconnect if the sever is not available
 #   at start or the servers crashes. This can be deactivated with this option
 #     --isaac.reconnect false
+#   Enable and write benchmark results into the given file.
+#     --isaac.timingsFilename benchRsults.txt
```
Comment (PrometheusPi) [inline docs/TBG_macros.cfg:292]:
```suggestion
#     --isaac.timingsFilename benchResults.txt
```

File: docs/source/usage/plugins/isaac.rst [hunk 2]
```diff
@@ -13,24 +13,28 @@ The plugin is available as soon as the :ref:`ISAAC library <install-dependencies
...
+Command line option               Description
+================================= ============================================================================
+``--isaac.period N``              Sets up, that every *N* th timestep an image will be rendered.
+                                  This parameter can be changed later with the controlling client.
+``--isaac.name NAME``             Sets the *NAME* of the simulation, which is shown at the client.
+``--isaac.url URL``               *URL* of the required and running isaac server.
+                                  Host names and IPs are supported.
+``--isaac.port PORT``             *PORT* of the isaac server.
+                                  The default value is ``2458`` (for the in-situ plugins), but may be needed
+                                  to be changed for tunneling reasons or if more than one server shall run on
```
Comment (PrometheusPi) [inline docs/source/usage/plugins/isaac.rst:26]:
```suggestion
                                  The default value is ``2458`` (for the in-situ plugins), but may needs to
                                  be changed for tunneling reasons or if more than one server shall run on
```

File: include/picongpu/param/isaac.param [hunk 3]
```diff
@@ -68,6 +68,10 @@ namespace picongpu
          *  the join of Native_Seq and Density_Seq. */
         using Fields_Seq = MakeSeq_t<Native_Seq, Density_Seq>;
 
+        /** Compile time sequence of all fields which shall be visualized. Basically
+         *  the join of Native_Seq and Density_Seq. */
```
Comment (PrometheusPi) [inline include/picongpu/param/isaac.param:72]:
```suggestion
         *  joining Native_Seq and Density_Seq. */
```

File: share/ci/install/isaac.sh [hunk 4]
```diff
@@ -19,7 +19,8 @@ sed -i 's/inline/GLM_FUNC_DECL/g' $GLM_ROOT/glm/gtc/type_ptr.inl
 cd $CI_PROJECT_DIR
 git clone https://github.com/ComputationalRadiationPhysics/isaac.git
 cd isaac
-git checkout 822f3c240a3bbdf3962984f4fc43f2ea7e75f2eb
+# issac version with bew LIC kernel https://github.com/ComputationalRadiationPhysics/isaac/pull/143
```
Comment (PrometheusPi) [inline share/ci/install/isaac.sh:22]:
```suggestion
# issac version with new LIC kernel https://github.com/ComputationalRadiationPhysics/isaac/pull/143
```

File: share/picongpu/benchmarks/SPEC/include/picongpu/param/isaac.param [hunk 5]
```diff
@@ -53,12 +53,25 @@ namespace picongpu
...
+        using Density_Seq = deriveField::CreateEligible_t<Particle_Seq, deriveField::derivedAttributes::Density>;
+        /** Intermediate list of filtered particle species, from which density fields
+         *  shall be created at runtime to visualize them.
+         *
+         *  You can create such densities from filtered particles by passing a particle
+         *  filter as the third template argument (`filter::All` by default). Don't forget
+         *  to add your filtered densities to the `Fields_Seq` below.
+         *  */
+        //  using Density_Seq_Filtered = deriveField::CreateEligible_t<Particle_Seq,
+        //      deriveField::derivedAttributes::Density, filter::All>;
```
Comment (PrometheusPi) [inline share/picongpu/benchmarks/SPEC/include/picongpu/param/isaac.param:65]: Is this comnented code OK?

File: share/picongpu/benchmarks/SPEC/include/picongpu/param/isaac.param [hunk 6]
```diff
@@ -53,12 +53,25 @@ namespace picongpu
...
 
         /** Compile time sequence of all fields which shall be visualized. Basically
          *  the join of Native_Seq and Density_Seq. */
         using Fields_Seq = MakeSeq_t<>;
 
+        /** Compile time sequence of all fields which shall be visualized. Basically
+         *  the join of Native_Seq and Density_Seq. */
```
Comment (PrometheusPi) [inline share/picongpu/benchmarks/SPEC/include/picongpu/param/isaac.param:72]:
```suggestion
         *  joining Native_Seq and Density_Seq. */
```

File: share/picongpu/tests/compileCurrentSolver/include/picongpu/param/isaac.param [hunk 7]
```diff
@@ -47,11 +47,31 @@ namespace picongpu
...
+         *  */
+        //  using Density_Seq_Filtered = deriveField::CreateEligible_t<Particle_Seq,
+        //      deriveField::derivedAttributes::Density, filter::All>;
 
         /** Compile time sequence of all fields which shall be visualized. Basically
          *  the join of Native_Seq and Density_Seq. */
         using Fields_Seq = MakeSeq_t<>;
 
+        /** Compile time sequence of all fields which shall be visualized. Basically
+         *  the join of Native_Seq and Density_Seq. */
```
Comment (PrometheusPi) [inline share/picongpu/tests/compileCurrentSolver/include/picongpu/param/isaac.param:72]:
```suggestion
         *  joining Native_Seq and Density_Seq. */
```

File: share/picongpu/tests/compileFieldSolver/include/picongpu/param/isaac.param [hunk 8]
```diff
@@ -47,11 +47,31 @@ namespace picongpu
...
+         *  */
+        //  using Density_Seq_Filtered = deriveField::CreateEligible_t<Particle_Seq,
+        //      deriveField::derivedAttributes::Density, filter::All>;
 
         /** Compile time sequence of all fields which shall be visualized. Basically
          *  the join of Native_Seq and Density_Seq. */
         using Fields_Seq = MakeSeq_t<>;
 
+        /** Compile time sequence of all fields which shall be visualized. Basically
+         *  the join of Native_Seq and Density_Seq. */
```
Comment (PrometheusPi) [inline share/picongpu/tests/compileFieldSolver/include/picongpu/param/isaac.param:72]:
```suggestion
         *  joining Native_Seq and Density_Seq. */
```

File: share/picongpu/tests/compileParticlePusher/include/picongpu/param/isaac.param [hunk 9]
```diff
@@ -47,11 +47,31 @@ namespace picongpu
...
+         *  */
+        //  using Density_Seq_Filtered = deriveField::CreateEligible_t<Particle_Seq,
+        //      deriveField::derivedAttributes::Density, filter::All>;
 
         /** Compile time sequence of all fields which shall be visualized. Basically
          *  the join of Native_Seq and Density_Seq. */
         using Fields_Seq = MakeSeq_t<>;
 
+        /** Compile time sequence of all fields which shall be visualized. Basically
+         *  the join of Native_Seq and Density_Seq. */
```
Comment (PrometheusPi) [inline share/picongpu/tests/compileParticlePusher/include/picongpu/param/isaac.param:72]:
```suggestion
         *  joining Native_Seq and Density_Seq. */
```

**Review by PrometheusPi — CHANGES_REQUESTED** [review]
A few typos. I provided a quick fix for all of them.

**Comment by PrometheusPi** [comment]
As discussed, text changes will be added in a separate pull request to not delay merging.

**Review by PrometheusPi — APPROVED** [review]

**Pull request merged**